### DISCUSSION TITLE
[Academic study] how do OSS developers annotate Technical Debt?

### DISCUSSION THREAD

**Discussion by TheDutchDevil** [WeblateOrg/weblate#6636] [General]
Hello,

We're a team of Software Engineering researchers investigating technical debt. Technical debt refers to ""not quite right code which we postpone making it right” (Cunningham) and this may often introduce a cost. More specifically, the aim of this study is to gather information on how open source developers respond to specific source code phenomena such as partially or wrongly implemented functionality. With the results of this study we aim to create a better understanding of how developers annotate issues in source code. These results will be used to support prioritisation of maintenance activities in software systems. The survey data will be collected, analyzed and stored so that research questions can be answered and the results can be published. Since the survey is anonymous, no personal information is collected. For the same reason we cannot modify or remove answers upon request.

If you contribute code to this project you can help us out by filling out the following survey: https://forms.office.com/r/xRg92faiNd. The survey should take you at most 15 minutes.

Thanks in advance,

Nathan

**Comment by kddartown** [comment]
Hi Nathan,

do you have any references to your research ?

I would be interested to read the findings.

Thanks

Gavin
